src/main/scala/org/broadinstitute/hail/driver/ExportVCF.scala

```diff
@@ -205,7 +205,57 @@ object ExportVCF extends Command {
           sb.append("GT:AD:DP:GQ:PL")
         gs.foreach { g =>
           sb += '\t'
-          sb.append(g)
+
+          sb.append(g.gt.map { gt =>
+            val p = Genotype.gtPair(gt)
+            s"${ p.j }/${ p.k }"
+          }.getOrElse("./."))
+
+          def appendOption(stringBuilder: StringBuilder, toAppend: Option[Array[Int]]): Unit = {
+            if (toAppend.isDefined)
+              toAppend.foreach(_.foreachBetween(sb.append(_))(sb += ','))
+            else
+              sb += '.'
+          }
+
+          def append[T <: AnyVal](stringBuilder: StringBuilder, toAppend: Array[T]): Unit = {
+            toAppend.foreachBetween(sb.append(_))(sb += ',')
```

Comment:
With a type parameter, this will call `StringerBuilder.append(Any)` which in turn calls toString allocating a String.  If you type specialize `append`, you can call the optimized routines like `StringBuilder.append(Int)`.